packages/insured-bridge-relayer/src/Relayer.ts

```diff
@@ -228,8 +212,59 @@ export class Relayer {
     }
   }
 
-  private async instantRelay(/* deposit: Deposit*/) {
-    // TODO: implement
+  private async instantRelay(deposit: Deposit, realizedLpFeePct: BN) {
+    await this.gasEstimator.update();
+    try {
+      const { receipt, transactionConfig } = await runTransaction({
+        web3: this.l1Client.l1Web3,
+        transaction: this.generateInstantRelayTx(deposit, realizedLpFeePct),
+        transactionConfig: { gasPrice: this.gasEstimator.getCurrentFastPrice().toString(), from: this.account },
+        availableAccounts: 1,
+      });
+      if (receipt.events)
+        this.logger.info({
+          at: "InsuredBridgeRelayer#Relayer",
+          type: "Relay instantly sent 🚀",
+          tx: receipt.transactionHash,
+          depositHash: receipt.events.RelaySpedUp.returnValues.depositHash,
+          instantRelayer: receipt.events.RelaySpedUp.returnValues.instantRelayer,
+          transactionConfig,
+        });
+      else throw receipt;
+    } catch (error) {
+      this.logger.error({ at: "InsuredBridgeRelayer#Relayer", type: "Something errored instantly relaying!", error });
+    }
+  }
+
+  private generateSlowRelayTx(deposit: Deposit, realizedLpFeePct: BN): TransactionType {
+    const bridgePool = this.l1Client.getBridgePoolForDeposit(deposit);
+    return (bridgePool.methods.relayDeposit(
+      deposit.depositId,
+      deposit.depositTimestamp,
+      deposit.l1Recipient,
+      deposit.l2Sender,
+      deposit.amount,
+      deposit.slowRelayFeePct,
+      deposit.instantRelayFeePct,
+      deposit.quoteTimestamp,
+      realizedLpFeePct
+    ) as unknown) as TransactionType;
+  }
+
+  private generateSpeedUpRelayTx(deposit: Deposit): TransactionType {
+    const bridgePool = this.l1Client.getBridgePoolForDeposit(deposit);
+    return (bridgePool.methods.speedUpRelay(deposit as any) as unknown) as TransactionType;
+  }
+
+  private generateInstantRelayTx(deposit: Deposit, realizedLpFeePct: BN): TransactionType {
```

Comment:
the magic of multicall. we can re-use the previous methods as well as instant relay is simply the combination of a slow and speed up relay :)